src/coreclr/jit/importer.cpp

```diff
@@ -3145,10 +3145,17 @@ void Compiler::impImportAndPushBox(CORINFO_RESOLVED_TOKEN* pResolvedToken)
         // and the other you get
         //    *(temp+4) = expr
 
-        if (opts.OptimizationDisabled())
+        // For minopts/debug code, try and minimize the total number
+        // of box temps by reusing an existing temp when possible.
+        const bool shareBoxedTemps =
+            opts.OptimizationDisabled() &&
+            // We don't share them for boxed Enum values to, potentially, avoid boxing in Enum.HasFlag
+            !((varTypeIsIntegral(exprToBox)) &&
+              (info.compCompHnd->isEnum(pResolvedToken->hClass, nullptr) == TypeCompareState::Must) &&
+              (!lvaHaveManyLocals()) && (!opts.compDbgCode) && (!opts.jitFlags->IsSet(JitFlags::JIT_FLAG_MIN_OPT)));
```

Comment:
This is very hard to read, and it would be better to do the JIT-EE call last.
Also I'm not sure `JitFlags::JIT_FLAG_MIN_OPT` is going to be the right check since it skips many of the JIT's own ways to switch to MinOpts. But I guess we can't really differentiate between "MinOpts because of IL thresholds" or "MinOpts because of tier 0" today.

I would at least write it as:

```suggestion
        // For minopts/debug code, try and minimize the total number
        // of box temps by reusing an existing temp when possible. However,
        // avoid sharing in some tier 0 cases to potentially, avoid boxing in Enum.HasFlag.
        const bool shareBoxedTemps =
            opts.OptimizationDisabled() &&
            (opts.compDbgCode || opts.jitFlags->IsSet(JitFlags::JIT_FLAG_MIN_OPT) || !varTypeIsIntegral(exprToBox) || lvaHaveManyLocals() || (info.compCompHnd->isEnum(pResolvedToken->hClass, nullptr) != TypeCompareState::Must));
```

But maybe it would be better to check for TIER0 flag.

Reply:
Yes, Tier0 flag would probably be better as Andy fixed it recently to not be set when debug/minopts is requested